### PR TITLE
systemd.go: Added watchdog metrics

### PR DESCRIPTION
**Pull request by jpds**

### CODE REVIEW

File: systemd/systemd.go [hunk 1]
```diff
@@ -186,6 +192,26 @@ func NewCollector(logger log.Logger) (*Collector, error) {
 		"Service unit egress IP accounting in packets.",
 		[]string{"name"}, nil,
 	)
+	watchdogEnabled := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "", "watchdog_enabled"),
```
Comment (SuperQ) [inline systemd/systemd.go:196]: Should we also declare a subsystem here?

```suggestion
const watchdogSubsystem = "watchdog"
...

		prometheus.BuildFQName(namespace, watchdogSubsystem, "enabled"),`
```

**Review by SuperQ — COMMENTED** [review]
Minor nit about use of namespace/subsystem.

Also, I'm not so sure about the use of a regexp here, seems like an extra/expensive step to parse a number.

**Comment by jpds** [comment]
I retested this on a prod box and we need the regex to remove the `@t` in the DBus value:

```
caller=systemd.go:317 level=debug msg="Failed to collect watchdog metrics" err="strconv.ParseFloat: parsing \"@t 149702468689\": invalid syntax"
```

File: systemd/systemd.go [hunk 2]
```diff
@@ -688,3 +726,73 @@ func (c *Collector) filterUnits(units []dbus.UnitStatus, includePattern, exclude
...
+	runtimeWatchdogUSecProperty, err := conn.GetManagerProperty("RuntimeWatchdogUSec")
+	if err != nil {
+		return err
+	}
+
+	parsedWatchdogLastPingMonotonic := watchdogTimeRE.FindString(watchdogLastPingMonotonicProperty)
+	parsedWatchdogLastPingTime := watchdogTimeRE.FindString(watchdogLastPingTimeProperty)
+	parsedRuntimeWatchdog := watchdogTimeRE.FindString(runtimeWatchdogUSecProperty)
+
+	watchdogLastPingMonotonic, err := strconv.ParseFloat(parsedWatchdogLastPingMonotonic, 64)
```
Comment (SuperQ) [inline systemd/systemd.go:766]:
```suggestion
	watchdogLastPingMonotonic, err := strconv.ParseFloat(strings.TrimLeft(watchdogLastPingMonotonicProperty, "@t "), 64)
```

**Review by SuperQ — APPROVED** [review]

**Pull request merged**